yb-voyager/src/migassessment/sizing.go

```diff
@@ -375,6 +389,125 @@ func pickBestRecommendation(recommendations map[int]IntermediateRecommendation)
+			if recCombined.VCPUsPerInstance > recAllSharded.VCPUsPerInstance {
+				selectedRec = recCombined
+				notSelectedRec = recAllSharded
+				reasoning = fmt.Sprintf("\nSelected colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores) over all-sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+					recCombined.VCPUsPerInstance, rec1ResultantCores, recAllSharded.VCPUsPerInstance, rec2ResultantCores)
+			} else {
+				selectedRec = recAllSharded
+				notSelectedRec = recCombined
+				reasoning = fmt.Sprintf("\nSelected all-sharded strategy with %d VCPUs per instance (%.1f resultant cores) over colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+					recAllSharded.VCPUsPerInstance, rec2ResultantCores, recCombined.VCPUsPerInstance, rec1ResultantCores)
+			}
+			vcpuConditionMet = true
+		} else if rec1ResultantCores < rec2ResultantCores {
+			selectedRec = recCombined
+			notSelectedRec = recAllSharded
+			reasoning = fmt.Sprintf("\nSelected colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores) over all-sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+				recCombined.VCPUsPerInstance, rec1ResultantCores, recAllSharded.VCPUsPerInstance, rec2ResultantCores)
+			vcpuConditionMet = true
+		} else if rec2ResultantCores < rec1ResultantCores {
+			selectedRec = recAllSharded
+			notSelectedRec = recCombined
+			reasoning = fmt.Sprintf("\nSelected all-sharded strategy with %d VCPUs per instance (%.1f resultant cores) over colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+				recAllSharded.VCPUsPerInstance, rec2ResultantCores, recCombined.VCPUsPerInstance, rec1ResultantCores)
+			vcpuConditionMet = true
+		}
+	}
+
+	// If VCPUsPerInstance condition was not met, use original node comparison logic
+	if !vcpuConditionMet {
+		// Compare cores needed - pick the one with fewer nodes, preferring all-sharded when equal
+		if recCombined.NumNodes < recAllSharded.NumNodes {
+			selectedRec = recCombined
+			notSelectedRec = recAllSharded
+			reasoning = fmt.Sprintf("\nSelected colocated+sharded strategy requiring %.0f nodes over all-sharded strategy requiring %.0f nodes. ",
+				recCombined.NumNodes, recAllSharded.NumNodes)
+		} else if recCombined.NumNodes > recAllSharded.NumNodes {
+			selectedRec = recAllSharded
+			notSelectedRec = recCombined
+			reasoning = fmt.Sprintf("\nSelected all-sharded strategy requiring %.0f nodes over colocated+sharded strategy requiring %.0f nodes. ",
+				recAllSharded.NumNodes, recCombined.NumNodes)
+		} else {
+			// Equal cores - prefer all-sharded (recAllSharded)
+			selectedRec = recAllSharded
+			notSelectedRec = recCombined
+			reasoning = fmt.Sprintf("\nSelected all-sharded strategy (same %.0f nodes as colocated+sharded strategy, preferring all-sharded). ",
+				recAllSharded.NumNodes)
+		}
+	}
+
+	// Get detailed object counts for non-selected recommendations
```

Comment:
if the "reasoning" info was just for debugging, can we convert all these to logs, instead of keeping them in the "reasoning" variable? 

or at least let's log the reasoning variable ?